packages/block-library/src/navigation-link/edit.js

```diff
@@ -129,6 +131,43 @@ function NavigationLinkEdit( {
 		selection.addRange( range );
 	}
 
+	function handleCreateEntity( type, entityTitle ) {
+		return saveEntityRecord( 'postType', type, {
+			title: entityTitle,
+			status: 'publish',
+		} ).then( ( entity ) => {
+			// `entity` may not reject the Promise
+			// but may still be invalid. Here we
+			// tests for unexpected values and throw accordingly.
+			if ( null === entity || undefined === entity ) {
+				throw new TypeError(
+					'API response returned invalid entity.',
+					entity
+				);
+			}
+
+			const requiredEntityProps = [ 'id', 'title', 'link' ];
+
+			const entityMissingProperty = requiredEntityProps.find(
+				( entityProp ) => ! entity.hasOwnProperty( entityProp )
+			);
+
+			if ( Boolean( entityMissingProperty ) ) {
+				throw new TypeError(
+					`API response returned invalid entity. Missing required property "${ entityMissingProperty }".`,
+					entity
+				);
+			}
+
+			return {
+				id: entity.id,
+				type,
+				title: entity.title.raw, // TODO: use raw or rendered?
```

Comment:
If `title` should be intended primarily for display, then I think `rendered` may be what you want.

```suggestion
				title: entity.title.rendered,
```